### DISCUSSION TITLE
How to debug the issue for Airflow DAG disappearing automatically from Webserver UI

### DISCUSSION THREAD

**Discussion by ashishmourya** [apache/airflow#31456] [Q&A]
Airflow DAG disappearing automatically from Webserver UI

We have multiple DAGs running at Airflow-2.3.2. One of particular DAG disappears automatically from the Webserver UI on random day (observed this issue couple of times). This DAG is having schedule_interval as None and runs on demand triggers externally. There is no change in DAG right before it disappears from the UI. DAG starts listing again immediately after a Webserver service restarts without any changes.

Tried following things -
- Webserver logs has no error captured.
- airflow dags list command is not showing this DAG

Any idea or steps to further debug or troubleshoot the issue would be very helpful.

Thanks!!

**Comment by potiuk** [comment]
It might not import properly depending on the code in your DAG and availabiity of something that the DAG needs to be parsed - for example if during parsing the DAG, the top level code uses  A DB or when it reaches out to pull something or uses a file locally that has been removed. Look at https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html#top-level-python-code

You should find some errors about it scheduler logs or dag file processor logs.
